projects/plugins/wpcomsh/feature-plugins/full-site-editing.php

```diff
@@ -87,16 +87,19 @@ function wpcom_public_coming_soon_replace_yoast_seo_notice() {
 	 */
 	$should_replace_yoast_notice = $are_search_engines_discouraged || $is_wpcom_public_coming_soon_enabled;
 
-	if ( $should_replace_yoast_notice && ! $wpseo_options['ignore_search_engines_discouraged_notice'] ) {
-		// Hide the Yoast SEO notice if it's not hidden and the site is set to Coming Soon mode.
-		$wpseo_options['ignore_search_engines_discouraged_notice'] = true;
-		update_option( 'wpseo', $wpseo_options );
-	}
+	if ( isset( $wpseo_options['ignore_search_engines_discouraged_notice'] ) ) {
+
+		if ( $should_replace_yoast_notice && ! $wpseo_options['ignore_search_engines_discouraged_notice'] ) {
```

Comment:
Probably this block should still be run if the key is unset, same as if the value is false.

Probably you could just use `empty()` here.